x-pack/plugins/security_solution/public/common/components/landing_page/onboarding/onboarding_with_settings.tsx

```diff
@@ -6,17 +6,23 @@
  */
 
 import React from 'react';
+import { useSpaceId } from '../../../hooks/use_space_id';
 import { useAvailableSteps } from './hooks/use_available_steps';
 import { useProductTypes } from './hooks/use_product_types';
 import { Onboarding } from './onboarding';
 
-export const OnboardingWithSettingsComponent: React.FC<{ indicesExist?: boolean }> = ({
+const OnboardingWithSettingsComponent: React.FC<{ indicesExist?: boolean }> = ({
   indicesExist,
 }) => {
   const productTypes = useProductTypes();
   const onboardingSteps = useAvailableSteps();
+  const spaceId = useSpaceId();
 
-  if (!onboardingSteps) {
+  /* spaceId returns undefined if the space is loading.
+   ** We render the onboarding component only when spaceId is ready
+   ** to make sure it reads the local storage data with the correct spaceId.
+   */
+  if (!onboardingSteps || !spaceId) {
```

Comment:
This fixes the problem, but it would be better if the `spaceId` value was kept in sync in the storage layer, we would not have to delay the page render until the space has loaded.

No need to do it now, but we should consider this change when we migrate the onboarding hub.